### PR TITLE
Garbage Collection and memory snapshot code for debugging partitioning pipeline

### PR DESCRIPTION
**Pull request by kylasa**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tools/distpartitioning/utils.py [hunk 1]
```diff
@@ -420,3 +422,26 @@ def get_idranges(names, counts):
...
+    GB = 1024 * 1024 * 1024
+    MB = 1024 * 1024
+
+    mem = psutil.virtual_memory()
+    avail = mem.available / MB
+    used = mem.used / MB
+    total = mem.total / MB
+
+    mem_string = '{:.0f} (MB) total, {:.0f} (MB) used, {:.0f} (MB) avail'.format(total, used, avail)
+    logging.debug(f'[Rank: {rank} MEMORY_SNAPSHOT] {mem_string} - {tag}')
```
Comment (frozenbugs) [inline tools/distpartitioning/utils.py:447]: Add a new line at the end of the file.

File: tools/distpartitioning/utils.py [hunk 2]
```diff
@@ -420,3 +422,26 @@ def get_idranges(names, counts):
...
+    """
+    GB = 1024 * 1024 * 1024
+    MB = 1024 * 1024
+
+    mem = psutil.virtual_memory()
+    avail = mem.available / MB
+    used = mem.used / MB
+    total = mem.total / MB
+
+    mem_string = '{:.0f} (MB) total, {:.0f} (MB) used, {:.0f} (MB) avail'.format(total, used, avail)
```
Comment (frozenbugs) [inline tools/distpartitioning/utils.py:446]: Try f-string to increase the readability?
https://realpython.com/python-string-formatting/#3-string-interpolation-f-strings-python-36

File: tools/distpartitioning/utils.py [hunk 3]
```diff
@@ -420,3 +422,26 @@ def get_idranges(names, counts):
 
     return tid_dict, gid_dict
 
+
+def mem_snapshot(tag, rank):
```
Comment (frozenbugs) [inline tools/distpartitioning/utils.py:426]: It is not a well-known convention to use mem for memory. Change it to memory_snapshot?

e.g. https://pytorch.org/docs/stable/generated/torch.cuda.memory_snapshot.html

File: tools/distpartitioning/convert_partition.py [hunk 4]
```diff
@@ -9,7 +9,9 @@
 import pandas as pd
 import constants
 from pyarrow import csv
-from utils import read_json, get_idranges
+from utils import read_json, get_idranges, mem_snapshot
+
+import gc
```
Comment (frozenbugs) [inline tools/distpartitioning/convert_partition.py:14]: Use isort to keep the import in order?
https://quip-amazon.com/J9cMAtI9NOuC/Python-Code-Style-Check#temp:C:KXU4172bb625aae4bb088505bd9e

Comment (kylasa) [reply tools/distpartitioning/convert_partition.py:14]: Done as suggested. will be included in a separate commit

File: tools/distpartitioning/utils.py (hunk 3 above)
Comment (kylasa) [reply tools/distpartitioning/utils.py:426]: Done as suggested.

File: tools/distpartitioning/utils.py (hunk 2 above)
Comment (kylasa) [reply tools/distpartitioning/utils.py:446]: Done as suggested.

File: tools/distpartitioning/utils.py (hunk 1 above)
Comment (kylasa) [reply tools/distpartitioning/utils.py:447]: Done as suggested.

**Comment by Rhett-Ying** [comment]
pls rebase it on master branch or the CI will always fail...

**Review by frozenbugs — APPROVED** [review]
LGTM with the code style, there still room to improve, but it is not blocking this PR.
Please get final approval from others before submit.

**Comment by Rhett-Ying** [comment]
moved to https://github.com/dmlc/dgl/pull/4598 and targets on `master` branch. Let's close this PR.

**Pull request closed without merging**